pennylane/tape/tape.py

```diff
@@ -1245,6 +1299,43 @@ def func(a):
 
         return self._measurements[0].numeric_type
 
+    @property
+    def _numeric_type_new(self):
+        """Returns the expected numeric type of the tape result by inspecting
+        its measurements.
+
+        Returns:
+            Union[type, Tuple[type]]: The numeric type corresponding to the result type of the
+            tape, or a tuple of such types if the tape contains multiple measurements
+
+        **Example:**
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=2)
+            a = np.array([0.1, 0.2, 0.3])
+
+            def func(a):
+                qml.RY(a[0], wires=0)
+                qml.RX(a[1], wires=0)
+                qml.RY(a[2], wires=0)
+
+            with qml.tape.QuantumTape() as tape:
+                func(a)
+                qml.state()
+
+        .. code-block:: pycon
+
+            >>> tape.numeric_type
+            complex
+        """
+        types = tuple(observable.numeric_type for observable in self._measurements)
```

Comment:
What happens in the shot-vector case? Do we want to make this method resemble `shape` such that the shot-vector structure is reflected in the output here?

Reply:
The `numeric_type` is device-independent, and we don't pass in the device to this attribute, so there's no way of knowing what the shot vector is.

If the concern is that the tuple structures of `shape` and `numeric_type` don't match, it's relatively straightforward to add a leading dimension to `numeric_type` in the shot-vector scenarios. See my updates to the jax-jit interface [here](https://github.com/PennyLaneAI/pennylane/blob/jax_jit_tap/pennylane/interfaces/jax_jit.py#L143).